TWTValidation/JSON Validator/AST Nodes/TWTJSONSchemaASTNode.h

```diff
@@ -29,20 +29,77 @@
+ @discussion This is not used for validation.
+ */
 @property (nonatomic, copy) NSString *schemaDescription;
+
+/*!
+ @abstract A flag indicating whether the "type" keyword is present in the schema.
+ @discussion A YES requires that an instance be of the type specified. A NO indicates that types not matching type-specific keywords will be ignored for those keywords.
+ */
 @property (nonatomic, assign, getter = isTypeSpecified) BOOL typeSpecified;
+
+/*!
+ @abstract A set of type keywords indicating the valid types for an instance described by the schema.
+ @discussion Subclasses must override this implementation with either a returned set or allowing readwrite access. May be nil, indicating type is not meaningful (e.g., dependencyNode).
+ */
 @property (nonatomic, copy, readonly) NSSet *validTypes;
-@property (nonatomic, copy) NSSet *validValues; //enum keyword
-@property (nonatomic, copy) NSArray *andSchemas; // allOf
-@property (nonatomic, copy) NSArray *orSchemas; // anyOf
-@property (nonatomic, copy) NSArray *exactlyOneOfSchemas; // oneOf
+
+/*!
+ @abstract A set of valid values given by "enum."
+ @discussion An instance passes if it is equal to one of these objects. Nil indicates "enum" is not present in the schema.
+ */
+@property (nonatomic, copy) NSSet *validValues;
+
+
+/*!
+ @abstract An array of TWTJSONSchemaASTNodes that model the schemas given by "allOf."
+ @discussion If this property is non-nil, it must have at least one element. Nil indicates "allOf" is not present in the schema.
+ */
+@property (nonatomic, copy) NSArray *andSchemas;
+
+/*!
+ @abstract An array of TWTJSONSchemaASTNodes that model the schemas given by "anyOf."
+ @discussion If this property is non-nil, it must have at least one element. Nil indicates "anyOf" is not present in the schema.
+ */
+@property (nonatomic, copy) NSArray *orSchemas;
+
+/*!
+ @abstract An array of TWTJSONSchemaASTNodes that model the schemas given by "oneOf."
+ @discussion If this property is non-nil, it must have at least one element. Nil indicates "oneOf" is not present in the schema.
+ */
+@property (nonatomic, copy) NSArray *exactlyOneOfSchemas;
+
+/*!
+ @abstract A TWTJSONSchemaASTNode that models the schema given by "not."
+ @discussion Nil indicates "not" is not present in the schema.
+ */
 @property (nonatomic, strong) TWTJSONSchemaASTNode *notSchema;
+
+/*!
+ @abstract A location where schemas referenced elsewhere can be defined.
+ @discussion Each key provides a reference path, and its value must be a TWTJSONSchemaASTNode.
+ */
 @property (nonatomic, copy) NSDictionary *definitions;
 
+/*!
+ @abstract The interface for accepting a TWTJSONSchemaASTProcessor.
+ @discussion Subclasses must override this, and the implementations must invoke the appropriate method from the TWTJSONSchemaASTProcessor protocol on the processor (for example, [processor processArrayNode:self]). If called on an instance of this class, this will throw a umk_subclassResponsibilityException.
```

Comment:
Add line breaks. Indent new lines with 4 spaces. Generally try to keep comments in the 130 character ballpack.